packages/curve-ui-kit/src/shared/ui/Metric.tsx

```diff
@@ -0,0 +1,193 @@
+import Box from '@mui/material/Box'
+import Skeleton from '@mui/material/Skeleton'
+import Typography from '@mui/material/Typography'
+
+import { SizesAndSpaces } from 'curve-ui-kit/src/themes/design/1_sizes_spaces'
+import { TypographyVariantKey, TYPOGRAPHY_VARIANTS } from 'curve-ui-kit/src/themes/typography'
+import { abbr, suffix } from 'curve-ui-kit/src/utils'
+
+const { Spacing } = SizesAndSpaces
+
+// Correspond to flexbox align items values.
+export const ALIGNMENTS = ['start', 'center', 'end'] as const
+type Alignment = (typeof ALIGNMENTS)[number]
+
+const MetricSize = {
+  small: 'highlightM',
+  medium: 'highlightL',
+  large: 'highlightXl',
+  extraLarge: 'highlightXxl',
+} as const satisfies Record<string, TypographyVariantKey>
+
+export const SIZES = Object.keys(MetricSize) as (keyof typeof MetricSize)[]
+
+type UnitOptions = {
+  symbol: string
+  position: 'prefix' | 'suffix'
+  abbreviate: boolean
+}
+
+const dollar: UnitOptions = {
+  symbol: '$',
+  position: 'prefix',
+  abbreviate: true,
+}
+
+const percentage: UnitOptions = {
+  symbol: '%',
+  position: 'suffix',
+  abbreviate: false,
+}
+
+export const UNITS = ['dollar', 'percentage'] as const
+type Unit = (typeof UNITS)[number] | UnitOptions
+
+const UNIT_MAP: Record<(typeof UNITS)[number], UnitOptions> = {
+  dollar,
+  percentage,
+} as const
+
+// Default value formatter.
+const formatValue = (value: number, decimals?: number): string => {
+  return value.toLocaleString(undefined, {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  })
+}
```

Comment:
Prettier doesn't support this. We can configure eslint with `"as-needed" enforces no braces where they can be omitted (default)` as per https://eslint.org/docs/latest/rules/arrow-body-style, but because of the fact that we use the same eslint config for different projects it means we have to convert quite a lot of arrow notation usages. I can do this if you want to, but I'd create a new PR for it as it's outside of the scope here.

Reply:
yeah 100% out of scope. I like the single-line functions but maybe not worth the effort/errors